### PR TITLE
feat: preview adt field when hover

### LINKED ISSUE #13977: [Feature Request] Preview struct fields when hover on type.
something like this:
image

### PR DESCRIPTION
**Pull request by wasd96040501**
Closes #13977 
20231108194345_rec_

### CODE REVIEW

**Comment by lnicola** [comment]
Nit: we should probably match the standard Rust formatting (`struct Foo {`).

File: crates/hir/src/display.rs [hunk 1]
```diff
@@ -163,7 +163,40 @@ impl HirDisplay for Struct {
         write!(f, "{}", self.name(f.db).display(f.db.upcast()))?;
         let def_id = GenericDefId::AdtId(AdtId::StructId(self.id));
         write_generic_params(def_id, f)?;
+
+        let field_is_tuple_index =
+            self.variant_data(f.db).fields().iter().any(|(_, f)| f.name.as_tuple_index().is_some());
```
Comment (lnicola) [inline crates/hir/src/display.rs:168]: Can we check `self.variant_data(f.db).kind()` here instead?

**Review by lnicola — APPROVED** [review]

**Comment by wasd96040501** [comment]
> Nit: we should probably match the standard Rust formatting (`struct Foo {`).

Sure. Fixes have been pushed.

File: crates/ide/src/hover/tests.rs [hunk 2]
```diff
@@ -2717,7 +2728,7 @@ fn main() { let s$0t = foo(); }
                                     focus_range: 52..54,
                                     name: "S2",
                                     kind: Struct,
-                                    description: "struct S2",
+                                    description: "struct S2 {\n}",
```
Comment (lnicola) [inline crates/ide/src/hover/tests.rs:2731]: These would normally be `struct S2 {}`, but feel free to ignore this if it's too much of a hassle.

**Comment by lnicola** [comment]
Let's test this little thing:

@bors2 try

**Comment by lnicola** [comment]
@bors2 help

**Comment by wasd96040501** [comment]
> Thanks, would you mind squashing?

Sure, already squashed.

**Comment by Veykril** [comment]
Related https://github.com/rust-lang/rust-analyzer/issues/3197

**Comment by Veykril** [comment]
I think we might want to expose a config for this, to enable, disable and maybe set a maximum field amount to render? Unsure, will probably see after having used this a bit.

**Pull request merged**